src/modules/erc20linear.ts

```diff
@@ -0,0 +1,54 @@
+// handler methods for the erc20 linear reward module
+
+import { Address, BigInt, log } from '@graphprotocol/graph-ts'
+import {
+  ERC20LinearRewardModule as ERC20LinearRewardModuleContract,
+} from '../../generated/templates/StakingModule/ERC20LinearRewardModule'
+import { RewardsFunded } from '../../generated/templates/RewardModule/ERC20BaseRewardModule'
+import { Staked, Unstaked, Claimed } from '../../generated/templates/StakingModule/ERC20StakingModule'
+import { Pool, Token, Position, } from '../../generated/schema'
+import { integerToDecimal } from '../util/common'
+import { ZERO_BIG_INT, ZERO_BIG_DECIMAL, ONE_E_18, INITIAL_SHARES_PER_TOKEN } from '../util/constants'
+
+
+export function handleRewardsFundedLinear(event: RewardsFunded, pool: Pool, token: Token): void {
+  let contract = ERC20LinearRewardModuleContract.bind(event.address);
+
+  // update timeframe for pool
+  if (pool.start.equals(ZERO_BIG_INT)) pool.start = event.params.timestamp;
+
+  // TODO do we still want to create an object here?
```

Comment:
Oh good point on the `Pool.state`, added some handling for that

Reply:
also a single `Funding` object could potentially be updated dynamically to just show current timestamp to end of runway. Do you think this would useful @IainM22 ? Downside is that we wouldn't see previous rate history

Reply:
I think its fine to skip all `Funding` since `Pool.end` can be used for tracking runway

Reply:
makes sense